NewDawn/NewDawn/MainPageViewController.swift

```diff
@@ -8,41 +8,108 @@
 
 import UIKit
 
-let MAIN_PAGE_PROFILE_INDEX = "main_page_profile_index"
 
 class MainPageViewController: UIViewController {
     
     @IBOutlet weak var tableView: UITableView!
     var viewModel: MainPageViewModel!
-    
-    func loadProfileIndex() -> Int {
-        if let index = self.localReadKeyValue(key: MAIN_PAGE_PROFILE_INDEX) as? Int {
-            return index
-        }
-        self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
-        return 0
-    }
-
-    func updateProfileIndex() -> Void {
-        self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: loadProfileIndex() + 1)
-    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        if loadProfileIndex() >= USER_DUMMY_DATA.count {
-            // TODO: When the user has seen all profiles, we go back to the first profile.
-            // In the future, we should go to an ending page
-            self.localStoreKeyValue(key: MAIN_PAGE_PROFILE_INDEX, value: 0)
+        self.getUserData(){ () -> () in
```

Comment:
It seems that each time user click skip, it will re-fetch profiles from backend. Ideally we should only fetch once when the main page is on daily fresh.
You can do that by moving this function to somewhere else such as checkMainPageReload() in ViewerController.swift. After the fetch, you can store the entire list of user profiles into local storage.